application/models/User.php

```diff
@@ -127,6 +127,19 @@ public function rules()
         );
     }
 
+    /** @inheritdoc */
+    public function scopes()
+    {
+        return array(
+            'active' => array(
+                'condition' => "expires > :now OR expires IS NULL",
+                'params' => array(
+                    'now' => dateShift(date("Y-m-d H:i:s"), "Y-m-d H:i:s", Yii::app()->getConfig("timeadjust")),
```

Comment:
Isn't there a common function to do this? If not, maybe we could add it and that wuld be the start of using it.
Disregard if you believe that is out of the scope.

Reply:
`dateShift` is the common function, no ?

Reply:
I meant something like `timeAdjustedDate()`.

Reply:
```
function configTimeAdjustedDate($date = null, $format = "Y-m-d H:i:s") {
    if (empty($date)) {
        $date = date("Y-m-d H:i:s");
    }
    return dateShift($date, $format, Yii::app()->getConfig("timeadjust")),
}
```

Really ?

Where ?

Reply:
I was saying to create that. Then, in other time, we can spread it

Reply:
https://github.com/LimeSurvey/LimeSurvey/pull/3525#discussion_r1358315823

Reply:
My opinion  it's not needed

Reply:
> My opinion it's not needed

What do you mean?
If you feel creating that function is too much, just don't. It was just a suggestion

Reply:
I think dateShift do the job.